shell/src/test/java/alluxio/cli/fsadmin/command/SummaryCommandTest.java

```diff
@@ -0,0 +1,193 @@
+    mPrintStream = new PrintStream(fileOutputStream);
+  }
+
+  @After
+  public void cleanDependencies() throws IOException {
+    mMetaMasterClient.close();
+    mBlockMasterClient.close();
+    mPrintStream.close();
+    mFile.delete();
+  }
+
+  @Test
+  public void callSummaryWithDependencies() throws IOException {
+    SummaryCommand summaryCommand = new SummaryCommand(mMetaMasterClient,
+        mBlockMasterClient, mPrintStream);
+    summaryCommand.run();
+    mPrintStream.flush();
+    mPrintStream.close();
+
+    BufferedReader reader = new BufferedReader(new FileReader(mFile));
+    checkIfOutputValid(reader);
+  }
+
+  /**
+   * Checks if the output contains expected values.
+   *
+   * @param reader reader to read SummaryCommand output
+   */
+  private void checkIfOutputValid(BufferedReader reader) throws IOException {
+    StringBuilder outputBuilder = new StringBuilder();
+    String line;
+    while ((line = reader.readLine()) != null) {
+      outputBuilder.append(line);
+    }
+    String output = outputBuilder.toString();
+
+    Assert.assertThat(output, CoreMatchers.containsString(
+        "Master Address: " + mAddress));
+    Assert.assertThat(output, CoreMatchers.containsString(
+        "Web Port: " + mWebPort));
+    Assert.assertThat(output, CoreMatchers.containsString(
+        "Rpc Port: " + mRpcPort));
+    Assert.assertThat(output, CoreMatchers.containsString(
+        "Started: " + CommonUtils.convertMsToDate(mStartTimeMs)));
+    Assert.assertThat(output, CoreMatchers.containsString(
+        "Uptime: " + CommonUtils.convertMsToClockTime(mUpTimeMs)));
+    Assert.assertThat(output, CoreMatchers.containsString(
+        "Version: " + mVersion));
+    Assert.assertThat(output, CoreMatchers.containsString(
+        "Safe Mode: " + mSafeMode));
+
+    Assert.assertThat(output, CoreMatchers.containsString("Live Workers: " + mLiveWorkerNum));
+    Assert.assertThat(output, CoreMatchers.containsString("Lost Workers: " + mLostWorkerNum));
+    Assert.assertThat(output, CoreMatchers.containsString("Total Capacity: "
+        + FormatUtils.getSizeFromBytes(mCapacityBytes)));
+    Assert.assertThat(output, CoreMatchers.containsString("Total Capacity: "
+        + FormatUtils.getSizeFromBytes(mCapacityBytes)));
+    Assert.assertThat(output, CoreMatchers.containsString("Used Capacity: "
+        + FormatUtils.getSizeFromBytes(mUsedBytes)));
+    Assert.assertThat(output, CoreMatchers.containsString("Free Capacity: "
+        + FormatUtils.getSizeFromBytes(mFreeBytes)));
+    Assert.assertThat(output, CoreMatchers.containsString("Tier: MEM"
+        + "  Size: " + FormatUtils.getSizeFromBytes(mCapacityBytesOnTiers.get("MEM"))));
+    Assert.assertThat(output, CoreMatchers.containsString("Tier: MEM"
+        + "  Size: " + FormatUtils.getSizeFromBytes(mUsedBytesOnTiers.get("MEM"))));
+  }
+
+  /**
+   * Sets the info values so that we can do explicitly check.
+   */
+  private void setInfoValues() {
```

Comment:
these could all be defined as `private static final` constants instead of fields at the top of the file

Reply:
changed. Thanks